server/src/main/java/org/elasticsearch/common/xcontent/ChunkedToXContentHelper.java

```diff
@@ -58,7 +58,7 @@ public static Iterator<ToXContent> xContentValuesMap(String name, Map<String, ?
         );
     }
 
-    public static Iterator<ToXContent> field(String name, boolean value) {
+    public static <T> Iterator<ToXContent> field(String name, T value) {
```

Comment:
Hm, by making this generic we are slightly changing the code because we will not be using the primitive `boolean` type but `Boolean`. Not sure, if this is a big issue, but it's something to consider. What about adding it instead of replacing the one that is already here?